table/builder.go

```diff
@@ -307,67 +323,54 @@ func (b *Builder) Finish() error {
 	}
 	b.w.Reset(idxFile)
 
-	var encodeBuf [8]byte
-
 	// Don't compress the global ts, because it may be updated during ingest.
 	ts := uint64(math.MaxUint64)
 	if b.isExternal {
 		// External builder doesn't append ts to the keys, the output sst should has a non-MaxUint64 global ts.
 		ts = math.MaxUint64 - 1
 	}
-	binary.BigEndian.PutUint64(encodeBuf[:], ts)
-	if err := b.w.Append(encodeBuf[:]); err != nil {
+	if err := b.w.Append(u64ToBytes(ts)); err != nil {
 		return err
 	}
 
-	binary.BigEndian.PutUint64(encodeBuf[:], uint64(b.writtenLen))
-	b.buf = append(b.buf, encodeBuf[:]...)
-
-	b.buf = append(b.buf, u32ToBytes(uint32(len(b.smallest)))...)
-	b.buf = append(b.buf, b.smallest...)
-	b.buf = append(b.buf, u32ToBytes(uint32(len(b.biggest)))...)
-	b.buf = append(b.buf, b.biggest...)
+	encoder := metaEncoder{b.buf}
 
-	b.buf = append(b.buf, u32ToBytes(uint32(len(b.baseKeysEndOffs)))...)
-	b.buf = append(b.buf, u32SliceToBytes(b.baseKeysEndOffs)...)
-	b.buf = append(b.buf, b.baseKeysBuf...)
-	b.buf = append(b.buf, u32SliceToBytes(b.blockEndOffsets)...)
+	encoder.appendUint(uint64(b.writtenLen), idFileSize)
```

Comment:
Is `idFileSize` the same as the last block end offset?

Reply:
Yes, but I think to save it explicitly is easier to understanding